lit_gpt/lora.py

```diff
@@ -373,25 +366,16 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             Output tensor of shape (batch_size, context_length, 3 * embedding_size)
         """
 
-        def T(w):
-            return w.T if self.fan_in_fan_out else w
```

Comment:
In `LoRALinear` for transposing `w.transpose(0,1)` is called, while for `LoRAQKVLinear` - `w.T`. 
So using the same transposing function for both classes might look like a wrong decision, because with `.T` method we are reversing the whole shape of a tensor.
But here we apply transposing function to only weights that are two dimensional, so it's all good. Annotations with shapes really help here.

If you feel uncomfortable about it I can override the function for `LoRAQKVLinear` like this:
```python
def T(self, x: torch.Tensor) -> torch.Tensor:
    """Transpose input tensor if weights are stored in format (fan_in, fan_out)"""
    return x.permute(*torch.arange(x.ndim -1, -1, -1)) if self.fan_in_fan_out else x
```
which looks convoluted because:
> UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matrices or `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor.

Reply:
The dimension is 2, so there should be no need to generalize to a form that is harder to read.

Reply:
Agree.
Just wanted to provide options.

Want to clarify: since the dimension is 2, `.transpose(0, 1)` == `.T`, thus we can reuse in `LoRAQKVLinear` method from `LoRALinear` and you are not against it, right?

Reply:
I would say `.T` and `.transpose` is what the average PyTorch user knows and can interpret without looking up the docs, so let's stick to these.